### PR TITLE
Document JavaConverters conversion from java.util.Properties to Map

### PR DESCRIPTION
**Pull request by janekdb**
Also
- Fix grammar on duplicated DecorateAsJava comment by copying over from
  JavaConverters
- Remove author tags

### CODE REVIEW

**Comment by janekdb** [comment]
An alternative approach is to delete the comment on `DecorateAsJava`.

**Comment by janekdb** [comment]
Review by @dickwall, @heathermiller, @som-snytt.

**Comment by janekdb** [comment]
Failed with,

```
[ERROR]
    /home/jenkins/workspace/scala-2.11.x-integrate-ide/target/scala-ide/org.scala-ide.sdt.core/
    src/org/scalaide/extensions/CompilerSupport.scala:98: error: value samePos is not a member
    of RichCompilationUnits.this.Tree
[ERROR]
    compilationUnitOfFile(file) exists (_.body.samePos(tree.pos))
```

File: src/library/scala/collection/convert/DecorateAsJava.scala [hunk 1]
```diff
@@ -53,10 +53,12 @@ import scala.language.implicitConversions
  *  - `scala.collection.Set` => `java.util.Set`
  *  - `scala.collection.Map` => `java.util.Map`
  *
- *  @author Martin Odersky
+ *  The following one way conversion is provided via `asScala`:
```
Comment (janekdb) [inline src/library/scala/collection/convert/DecorateAsJava.scala:56]: `DecorateAsJava` does not contain this but it doesn't contain any `asScala` methods even though line 20 notes that it does. More evidence this comment should be deleted.

**Comment by SethTisue** [comment]
LGTM

**Pull request merged**